v2/tracing.go

```diff
@@ -5,13 +5,18 @@ import (
 
 	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus"
 	"github.com/Azure/go-shuttle/v2/otel"
+	"go.opentelemetry.io/otel/trace"
 )
 
 // NewTracingHandler is a shuttle middleware that extracts the context from the message Application property if available,
 // or from the existing context if not, and starts a span.
-func NewTracingHandler(next Handler) HandlerFunc {
+func NewTracingHandler(next Handler,
+	extractFuncs ...func(ctx context.Context, message *azservicebus.ReceivedMessage) (context.Context, trace.Span)) HandlerFunc {
+	if len(extractFuncs) == 0 {
+		extractFuncs = append(extractFuncs, otel.Extract)
+	}
 	return func(ctx context.Context, settler MessageSettler, message *azservicebus.ReceivedMessage) {
-		ctx, span := otel.Extract(ctx, message)
+		ctx, span := extractFuncs[len(extractFuncs)-1](ctx, message)
```

Comment:
Why getting the last item in `extractFuncs[]`: From the current usage, I am not sure if creating a new span for each `extractFunc` in `extractFuncs` is a good idea, plus using `defer` in for loop is discouraged, so the overall logic will be changed more than I'd like. There is also no use case yet to pass in multiple `extractFunc`, so I decided to just stick to current logic and just get the last `extractFunc`.

Reply:
>I am not sure if creating a new span for each extractFunc in extractFuncs is a good idea

definitely not wanted.
My suggestion was to allow the user to enrich the span that is already extracted. We do not want to let the consumer control _how_ the span is _extracted_ from the message, because that would require them to also know how the span is _propagated_ (where it is stored on the message).

exposing an enrich func allows the framework to control the span extraction, but lets the user pull additional information to enrich the span, which will then cascade down in the trace.